src/Interpreters/parseColumnsListForTableFunction.cpp

```diff
@@ -48,7 +48,7 @@ void validateDataType(const DataTypePtr & type, const DataTypeValidationSettings
 
     if (!settings.allow_suspicious_fixed_string_types)
     {
-        auto basic_type = removeLowCardinality(removeNullable(type));
+        auto basic_type = removeLowCardinalityAndNullable(type);
```

Comment:
Here and in [InterpreterCreateQuery.cpp](https://github.com/ClickHouse/ClickHouse/pull/55479/files#diff-da014867807d23b4f33a776259af48f88e03f9683f334cceed5773b51af54edb) it was misused. I am not sure if it was affecting anything, though.